### PR TITLE
Preprocess data

### PR DESCRIPTION
**Pull request by otrnovec**

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
+/workspace.xml
```
Comment (janrygl) [inline .idea/.gitignore:3]: please create file
`.gitignore`
and put stuff into it like
```
.idea
*.pyc
__pycache__
```

After that, remove idea files from git and push:
`git rm --cached .idea`

Thanks

File: data/PARO_original.csv [hunk 2]
```diff
@@ -0,0 +1,1065 @@
+﻿properties.id;properties.name;properties.proposer;properties.status.id;properties.status.name;properties.district;properties.year;properties.budget;properties.detail;properties.category.name;properties.image;latitude;longitude;category_category;category.status;ObjectId;x;y
```
Comment (janrygl) [inline data/PARO_original.csv:1]: I recommend to add data/README.md file to folder and explain content of each csv file in one line, e.g.

 - PARO_original.csv: stores ...

Also, recommend to lowercase names, e.g. paro_original.csv

File: preprocess_data.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
+import numpy as np
```
Comment (janrygl) [inline preprocess_data.py]: add requirements.txt file to project naming all non-native packages, please

File: preprocess_data.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+# plt.show()
+
+# correlation heatmap
+sns_plot = sns.heatmap(df.corr(numeric_only=True), cmap="rocket_r")
+plt.xticks(rotation=15)
+plt.show()
+
+print(df.info())
+
+df.to_csv("data/paro_preprocessed.csv", sep=";", encoding="utf-8")
```
Comment (janrygl) [inline preprocess_data.py]: we need to handle paths properly if the script is called from somewhere else

new_path = Path(__file__).parent / "data" / "paro_preprocessed.csv"
uses relative path to this file

or you can define folder in some settings.py script

**Review by janrygl — COMMENTED** [review]
Thank you for your code, we can discuss in lessons

**Review by Dolezal-557280 — COMMENTED** [review]
The code seems clean, well structured and with effiecinet descriptions providing enough information about its different parts.

File: preprocess_data.py (hunk 4 above)
Comment (janmalyy) [reply preprocess_data.py]: done

File: preprocess_data.py (hunk 3 above)
Comment (janmalyy) [reply preprocess_data.py]: done

File: data/PARO_original.csv (hunk 2 above)
Comment (janmalyy) [reply data/PARO_original.csv:1]: done

File: .idea/.gitignore (hunk 1 above)
Comment (janmalyy) [reply .idea/.gitignore:3]: done in different branch:)

**Pull request merged**